src/shared/vscode/commands2.ts

```diff
@@ -306,34 +302,86 @@ interface CommandInfo<T extends Callback> {
     readonly logging?: boolean
     /** Does the command require credentials? (default: false) */
     readonly autoconnect?: boolean
-    readonly errorHandler?: (error: Error) => ReturnType<T> | void
+}
+
+// This debouncing is a temporary safe-guard to mitigate the risk of instrumenting all
+// commands with telemetry at once. The alternative is to do a gradual rollout although
+// it's not much safer in terms of weeding out problematic commands.
+const emitInfo = new Map<string, { token: number; startTime: number; debounceCounter: number }>()
+const emitTokens: Record<string, number> = {}
+
+function startRecordCommand(id: string): number {
+    const currentTime = Date.now()
+    const previousEmit = emitInfo.get(id)
+    const threshold = isAutomation() ? 0 : 60000
```

Comment:
should we use the telemetry flush period here, at least as a placeholder? https://github.com/aws/aws-toolkit-vscode/blob/4bd84c2a729dccdea48354a7b40f86a627a1b070/src/shared/telemetry/telemetryService.ts#L30